content/spin/csharp-components.md

```diff
@@ -0,0 +1,100 @@
+title = "Building Spin Components in C#"
+template = "spin_main"
+date = "2022-03-14T00:22:56Z"
+[extra]
+
+---
+- [Prerequisites](#prerequisites)
+- [C# HTTP Handler](#c-http-handler)
+
+C# is one of the most widely used programming languages. Therefore, Spin provides support for building WebAssembly (Wasm) components in C# through an experimental .NET SDK. This allows you to build web applications, send HTTP requests and read/update databases. 
+
+This guide assumes you are familiar with the C# programming language and the concept of WebAssembly. If you are interested in reading more about .NET and Wasm please also see our blog article [WebAssembly for .NET Developers: Introducing the Spin .NET SDK](https://www.fermyon.com/blog/webassembly-for-dotnet-developers-spin-sdk-intro).
+
+### Prerequisites
+
+To create a new .NET application, you’ll first need to install .NET 7 Preview 5 or above, Wizer and the Spin `http-csharp` template. Ensure Wizer is on your place or install it using `cargo install wizer --all-features`. Additionally, ensure you have the latest version of [spin installed](./install.md).
+
+Install the `http-csharp` templates as follows:
+
+```bash
+spin templates install --git https://github.com/fermyon/spin-dotnet-sdk --branch main --update
+```
+
+If you already have this template installed, you can upgrade it with `spin templates upgrade`. 
+
+Then to create an application based on this template, input the following into your chosen repository:
+
+```bash
+spin new http-csharp
+```
+
+After following resulting prompts, you should now have an application repository that includes a `spin.toml`, `handler.cs`, `Project.csproj` file and more.
+
+To build and run the initially scaffolded application simply run `spin build --up`. 
+
+### C# HTTP Handler
+
+In Spin, a trigger handler - such as an HTTP request handler - is a function that receives event data as a structured argument. In traditional C# terms, it’s a public method in a DLL.
+
+To tell Spin which function in the DLL is the handler entry point, the SDK provides the `HttpHandler` attribute. An HTTP handler function looks like this:
+
+```csharp
+using System.Net;
+using Fermyon.Spin.Sdk;
+
+namespace Microservice;
+
+public static class Handler
+{
+    [HttpHandler]
+    public static HttpResponse HandleHttpRequest(HttpRequest request)
+    {
+        return new HttpResponse
+        {
+            StatusCode = HttpStatusCode.OK,
+            BodyAsString = "Hello from .NET",
+        };
+    }
+}
+```
+
+Your custom function can accept a reasonable amount of .NET code/syntax (subject to current WASI limitations). Here’s an example of a C# function that echoes request information:
```

Comment:
"a reasonable amount" makes it sound like there's a limit on how much code I can put into the function.  I think you intend to indicate that only a subset of .NET libraries/types/functions are supported?

Also this feels like an odd comment to combine with "Here's an example".  It leads the reader to interpret it as an example of the "reasonable amount."

What if we moved the stuff about limitations to a separate heading or a note block further down in the post?